src/main/scala/iog/psg/cardano/jpi/HelpExecute.scala

```diff
@@ -25,6 +25,14 @@ object HelpExecute {
       case (k, v) => k.toLong -> MetadataValueStr (v)
     }
   }.toMap
+
+  def failOnLeft[T](future: Future[CardanoApiResponse[T]])(implicit ec: ExecutionContext): Future[T] = for {
```

Comment:
failOnLeft should be unwrap and since we're in the area future should probably be responseF